src/Features/Core/Portable/EmbeddedLanguages/Json/LanguageServices/JsonClassifier.cs

```diff
@@ -187,7 +197,7 @@ public void Visit(JsonTextNode node)
 
             public void Visit(JsonCommaValueNode node)
             {
-                AddClassification(node.CommaToken, ClassificationTypeNames.JsonPunctuation);
+                // already handled when we recurse in AddTokenClassifications
```

Comment:
I don't think I understand how the logic in here works. Why are commas special versus other types of tokens handled by the other visit methods?

Reply:
Commas are handled slightly specially in json (for error recovery purposes). They can both be a node (with a token in them), or just a token in separated list. We were only handling the first case. 

Now we handle both cases by just checking the token case only.

Reply:
Probably worth documenting that :)

Reply:
That's what I thought that comment was :'(

Reply:
This comment explains a bit of the why, but not the "commas are special for error recovery" part. That's what I needed to know to understand the code 🙂.